Classes/Plugin/PageView.php

```diff
@@ -219,6 +232,59 @@ protected function getFulltext($page) {
         return $fulltext;
     }
 
+    /**
+     * Get all AnnotationPages / AnnotationLists that contain text Annotations with "painting" motivation
+     *
+     * @param integer    $page: the current page's number
+     * @return array     An array containing the IRIs of the AnnotationLists / AnnotationPages as well as
+     *                   some information about the canvas.
+     */
+    public function getAnnotationContainers($page)
+    {
+        if ($this->doc instanceof IiifManifest) {
+            $canvasId = $this->doc->physicalStructure[$page];
+            $iiif = $this->doc->getIiif();
+            if ($iiif instanceof ManifestInterface) {
+                $canvas = $iiif->getContainedResourceById($canvasId);
+                /* @var $canvas \Ubl\Iiif\Presentation\Common\Model\Resources\CanvasInterface */
+                if ($canvas!=null && !empty($canvas->getPossibleTextAnnotationContainers(Motivation::PAINTING))) {
+                    $annotationContainers = array();
+                    /*
+                     *  TODO Analyzing the annotations on the server side requires loading the annotation lists / pages
+                     *  just to determine wether they contain text annotations for painting. This will take time and lead to a bad user experience.
+                     *  It would be better to link every annotation and analyze the data on the client side.
+                     *
+                     *  On the other hand, server connections are potentially better than client connections. Downloading annotation lists
+                     */
```

Comment:
Annotations could be anything, OCR fulltext as well as descriptions, for example. It's common to provide OCR fulltext as annotations with the `motivation` [`painting`](http://iiif.io/api/presentation/2#painting), but it can't be assumed that textual annotations with `motivation` `painting` are always fulltext. That's why I introduced the `AnnotationTool`. It currently displays only plain text annotations with the `motivation` `painting`, but I'd like to extent this to all motivations, with either a configuration option for the Toolbox plugin or the extension, or with a motivation selector within the AnnotationTool, with `painting` as default, if this is desired.
A check for the existence of annotations shouldn't be done for the whole `Document` because annotations are linked via the single `Canvas`. I would suggest to introduce a check in `Ubl\Iiif\Presentation\Common\Model\Resources\Canvas` with a one time lazy loading of the `AnnotationList` / `AnnotationPage`, but leave this code as it is for this PR.